### ISSUE TITLE
Benchmark accuracy performance dropping sharp with increased image size

### ISSUE THREAD

**Issue by nkucza** [ultralytics/ultralytics#23741]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Other

### Bug

We are experimenting with Yolo26 and different image sizes for detecting small objects. In our case, we see the expected plateauing of the accuracy. For sanity reasons, I tried the same with your benchmark function, and the opposite is the case. Increasing the input image size decreases the mAP50-95.
Here is an example what we see:

input size   : mAP50-95
640x640     : 0.6556
1024x1024 :  0.5563
1408x1408 :  0.427

I also checked it with the formats of PyTorch, ONNX, and TensorRT, and settings like nms.
Am I in the wrong, that that's not what should happen?

### Environment

Ultralytics 8.4.14 🚀 Python-3.10.12 torch-2.10.0a0+git449b176 CUDA:0 (Orin, 30697MiB)
Setup complete ✅ (12 CPUs, 30.0 GB RAM, 33.3/56.6 GB disk)

OS                     Linux-5.15.148-tegra-aarch64-with-glibc2.35
Environment            Linux
Python                 3.10.12
Install                pip
Path                   /mnt/storage/enableato/venv_eato_torch_2_10/lib/python3.10/site-packages/ultralytics
RAM                    29.98 GB
Disk                   33.3/56.6 GB
CPU                    ARMv8 Processor rev 1 (v8l)
CPU count              12
GPU                    Orin, 30697MiB
GPU count              1
CUDA                   12.6

numpy                  ✅ 2.2.6>=1.23.0
matplotlib             ✅ 3.10.7>=3.3.0
opencv-python          ❌ (not installed)>=4.6.0
pillow                 ✅ 12.0.0>=7.1.2
pyyaml                 ✅ 5.4.1>=5.3.1
requests               ✅ 2.25.1>=2.23.0
scipy                  ✅ 1.15.3>=1.4.1
torch                  ✅ 2.10.0a0+git449b176>=1.8.0
torch                  ✅ 2.10.0a0+git449b176!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.25.0>=0.9.0
psutil                 ✅ 5.9.0>=5.8.0
polars                 ✅ 1.38.1>=0.20.0
ultralytics-thop       ✅ 2.0.18>=2.0.18
{'OS': 'Linux-5.15.148-tegra-aarch64-with-glibc2.35', 'Environment': 'Linux', 'Python': '3.10.12', 'Install': 'pip', 'Path': '/mnt/storage/enableato/venv_eato_torch_2_10/lib/python3.10/site-packages/ultralytics', 'RAM': '29.98 GB', 'Disk': '33.3/56.6 GB', 'CPU': 'ARMv8 Processor rev 1 (v8l)', 'CPU count': 12, 'GPU': 'Orin, 30697MiB', 'GPU count': 1, 'CUDA': '12.6', 'Package Info': {'numpy': '✅ 2.2.6>=1.23.0', 'matplotlib': '✅ 3.10.7>=3.3.0', 'opencv-python': '❌ (not installed)>=4.6.0', 'pillow': '✅ 12.0.0>=7.1.2', 'pyyaml': '✅ 5.4.1>=5.3.1', 'requests': '✅ 2.25.1>=2.23.0', 'scipy': '✅ 1.15.3>=1.4.1', 'torch': '✅ 2.10.0a0+git449b176!=2.4.0,>=1.8.0; sys_platform == "win32"', 'torchvision': '✅ 0.25.0>=0.9.0', 'psutil': '✅ 5.9.0>=5.8.0', 'polars': '✅ 1.38.1>=0.20.0', 'ultralytics-thop': '✅ 2.0.18>=2.0.18'}}

opencv-python is installed.. compiled by hand. don`t know why it is marked here

### Minimal Reproducible Example

just use the function 

benchmark(model=yolo_weight_path, data="coco128.yaml", nms=True, imgsz= resolution[1], half=use_fp16, device="cuda:0", format="onnx", batch=1)

with different sizes

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @nkucza, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

This is an automated response 🤖—an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. In your case, it would be helpful to include a fully copy-pastable snippet (including how `model` is loaded, exact `resolution` values used, and the full command/output), plus the exact versions/settings used for each export format (PyTorch/ONNX/TensorRT) so we can reproduce the mAP50-95 drop across `imgsz` changes ✅

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Increasing image size wouldn't necessarily increase mAP. The model is good at the object sizes it was trained at. Increasing image size only would help if increasing it brings the objects closer to the size that they appear in the training set.

**Comment by nkucza** [comment]
I know that it is trained in a smaller input size and can be used as is, but increasing the input size should definitely not reduce the mAP as it does here.

**Comment by Y-T-G** [comment]
It depends on your dataset. You're making large objects even larger which will hurt mAP because the model didn't learn to detect them at that size. You're testing on COCO128 which has many close-up/large objects in images.

**Comment by Y-T-G** [comment]
Anything that moves inference away from training distribution hurts mAP. Even if it's as simple as changing the interpolation method used during resizing
